pinot-common/src/main/java/com/linkedin/pinot/common/utils/FileUploadUtils.java

```diff
@@ -94,11 +95,12 @@ public EntityEnclosingMethod forUri(String uri) {
 
   public static int sendFile(final String host, final String port, final String path, final String fileName,
       final InputStream inputStream, final long lengthInBytes, SendFileMethod httpMethod) {
-    return sendFile("http://" + host + ":" + port + "/" + path, fileName, inputStream, lengthInBytes, httpMethod);
+    return sendFile("http://" + host + ":" + port + "/" + path, fileName, inputStream, lengthInBytes,
+        httpMethod, null);
   }
 
   public static int sendFile(String uri, final String fileName, final InputStream inputStream, final long lengthInBytes,
-      SendFileMethod httpMethod) {
+      SendFileMethod httpMethod, String crc) {
```

Comment:
suggest rename 'crc' to ifMatchSegmentCrc to clarify what it signifies.

Also, please add some comments that this is useful only for segment push to controller. This class seems to be a generic FileUpload from anywhere to anywhere...

Lastly, the type of ifMatchSegmentCrc should be long, not string

Reply:
suggest adding a Header[] argument instead of crc